nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/service/loading.service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, delay, Observable } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class LoadingService {
+    status$: Observable<boolean>;
+
+    loading: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
```

Comment:
Hey @scottyaslan, thanks for the comments! The majority rxjs usage is around ngrx. As signal support within ngrx was still taking shape when this effort started, rxjs was used. With using ngrx as our store throughout the application, there are only a couple instances where we've needed to manage our own subjects. Also, takeUntilDestroyed has made the unsubscribe strategy simple.

All that said signals look compelling. I'm definitely happy to consider what makes most sense for NiFi as new features become widely available. And once this migration is complete, it will make future incremental changes way more manageable than they were previously.

Reply:
Absolutely, I completely agree with you on the effectiveness of using ngrx for the store pattern. Your point about reducing the instances where we manage our own subjects makes a lot of sense. I'm all for keeping the ngrx store within components, and the idea of leveraging these new Angular signals sounds like a great way to streamline our approach otherwise.

By the way, have you had a chance to explore ngrx-data? I came across it recently, and it seems like a promising open-source package with an MIT license. It's designed for handling API interactions (GET, POST, PUT, DELETE, and more) and seamlessly integrating the retrieved data into the store. Here's the link for reference: https://v8.ngrx.io/guide/data. I'd love to hear your thoughts on whether it could complement our current setup.

Reply:
Thanks for the pointer to ngrx-data. It looks like this sub-project was recently put in maintenance mode [1]. 

[1] https://ngrx.io/guide/data